lib/matplotlib/axes/_axes.py

```diff
@@ -2734,16 +2734,13 @@ def stem(self, *args, linefmt=None, markerfmt=None, basefmt=None, bottom=0,
             raise TypeError('stem expected between 1 and 5 positional '
                             'arguments, got {}'.format(args))
 
-        y = np.asarray(args[0])
```

Comment:
Not sure if you need this `asarray` bit? I guess it's covered by `convert_xunits`?

Reply:
Yes; in fact this change means that stem now correctly handles units on y-data (whereas they would be previously dropped by the call to asarray).  Not an intentional change though :)